packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -277,6 +278,51 @@ Amazon SES, however, support for Amazon SES is not available in the CDK yet. If
 give [this issue](https://github.com/aws/aws-cdk/issues/6768) a +1. Until then, you can use the [cfn
 layer](https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html) to configure this.
 
+### Lambda Triggers
+
+User pools can be configured such that AWS Lambda functions can be triggered when certain user operations or actions
+occur, such as, sign up, user confirmation, sign in, etc. They can also be used to add custom authentication
+challenges, user migrations and custom verification messages. Learn more about triggers at [User Pool Workflows with
+Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html).
+
+Lambda triggers can either be specified as part of the `UserPool` construct's construction, or it can be added later,
+via methods on the construct, as so -
+
+```ts
+const authChallengeFn = new lambda.Function(this, 'authChallengeFn', {
+  // ...
+});
+
+const userpool = new UserPool(this, 'myuserpool', {
+  // ...
+  triggers: {
+    createAuthChallenge: authChallengeFn,
+    // ...
+  }
+});
+
+userpool.addUserMigrationTrigger(new lambda.Function(this, 'userMigrationFn', {
+  // ...
+}));
+```
+
+The following table lists the set of triggers available, and their corresponding method to add it to the user pool.
+For more information on the function of these triggers and how to configure them, read [User Pool Workflows with
+Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html).
+
+| Trigger | Method |
+| ---- | ---- |
+| createAuthChallenge | `addCreateAuthChallengeTrigger()` |
+| customMessage | `addCustomMessageTrigger()` |
+| defineAuthChallenge | `addDefineAuthChallengeTrigger()` |
+| postAuthentication | `addPostAuthenticationTrigger()` |
+| postConfirmation | `addPostConfirmationTrigger()` |
+| preAuthentication | `addPreAuthenticationTrigger()` |
+| preSignUp | `addPreSignUpTrigger()` |
+| preTokenGeneration | `addPreTokenGenerationTrigger()` |
+| userMigration | `addUserMigrationTrigger()` |
+| verifyAuthChallengeResponse | `addVerifyAuthChallengeResponseTrigger()` |
+
```

Comment:
Would have been nice if add the trigger methods had a common prefix, and actually, since basically all of these have the exact same signature, perhaps we can consolidate them into a single method:

```ts
pool.addTrigger(Event.CreateAuthChallenge, fn);
```

I can see the value in different signatures if each trigger had some kind of specialized signature, but it's basically like a generic event handler, no?